### PR TITLE
Add initial lowering of aten depthwise convolution to tosa.depthwise_conv2d

### PR DESCRIPTION
**Pull request by tatwaichong**
aten.convolution is known as depthwise convolution when groups == in_channels and out_channels == K * in_channels. Lower this operation types to tosa.depthwise_conv2d.

### CODE REVIEW

**Comment by tatwaichong** [comment]
@digantdesai, No one assigned to this poor PR, can ask you to take a look of it?

**Comment by tatwaichong** [comment]
Rebase.

**Comment by digantdesai** [comment]
> @digantdesai, No one assigned to this poor PR, can ask you to take a look of it?

Feel free to assign it to me and add `Partner: Arm` label to the PR. 👍

File: backends/arm/operators/op_avg_pool2d.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def define_node(
         # Transpose to align with TOSA
         NHWC_Order = [0, 2, 3, 1]
         input_transposed = transpose_helper(
-            tosa_graph, input_tensor, NHWC_Order, output.dtype
+            tosa_graph, input_tensor.name, input_tensor.shape, NHWC_Order, output.dtype
```
Comment (digantdesai) [inline backends/arm/operators/op_avg_pool2d.py:55]: unrelated change?

**Review by digantdesai — APPROVED** [review]
LGTM Thanks @taivu1998

File: examples/arm/arm_tosa_e2e.py [hunk 2]
```diff
@@ -46,8 +46,13 @@
     "simple_conv2d_2x2_1x1x14x14_stride2",
     "simple_conv2d_5x5_3x2x128x128_stride1",
     "simple_conv2d_2x2_3x1x40x40_non_bias",
+    "simple_depthwise_conv2d_3x3x3_1x3x256x256_group3_stride1",
+    "simple_depthwise_conv2d_8x4x3x3_1x4x256x256_group4_stride1",
+    "simple_depthwise_conv2d_3x3_1x4x256x256_group4_non_bias",
+    "simple_depthwise_conv2d_8x16x3_2x8x198x198_group8_stride3",
```
[the remark this answers is not in the document]
Comment (Jerry-Ge) [reply examples/arm/arm_tosa_e2e.py:52]: yep, added this to the next TODO

File: backends/arm/test/test_models.py [hunk 3]
```diff
@@ -336,19 +340,137 @@ def forward(self, x):
...
         def __init__(self):
             super().__init__()
-            self.conv2d = torch.nn.Conv2d(
+            """ in_channels == out_channels """
+            self.depthwise_conv2d = torch.nn.Conv2d(
                 in_channels=3, out_channels=3, kernel_size=3, stride=1, groups=3
             )
+            with torch.no_grad():
+                self.depthwise_conv2d.weight.copy_(
+                    rand_test_integers(low=100, high=130, size=(3, 1, 3, 3))
```
[the remark this answers is not in the document]
Comment (tatwaichong) [reply backends/arm/test/test_models.py:358]: The value distribution generated from `torch.randn` is too wide, and that makes the result from quantized model hard to match it's floating point counterpart in a nice precision difference.

Instead currently set a reasonable numerical range for testing (torch.randn looks like doesn't support this). BTW, `rand_test_integers` is just a kind of syntactic sugar for this cumbersome invocation chain. i.e. `torch.from_numpy(np.float32(rng.integers(low, high, size)))`

File: backends/arm/operators/op_avg_pool2d.py (hunk 1 above)
Comment (tatwaichong) [reply backends/arm/operators/op_avg_pool2d.py]: Nice catch. This change is unnecessary. Tidy it up in the following patch.

File: examples/arm/arm_tosa_e2e.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tatwaichong) [reply examples/arm/arm_tosa_e2e.py]: Got you. I realized adding these tests in the list of `exeample/arm/run.sh` is unnecessary.
Since these new depthwise conv2d tests are tested via `backends/arm/test/test_tosa.py` under the CI. is this right?

**Comment by digantdesai** [comment]
Is this good to go?
